backend/api/MQTT/MqttService.cs

```diff
@@ -0,0 +1,186 @@
+﻿using System.Text.Json;
+using Api.MQTT.Events;
+using Api.MQTT.MessageModels;
+using Api.Utilities;
+using MQTTnet;
+using MQTTnet.Client.Connecting;
+using MQTTnet.Client.Disconnecting;
+using MQTTnet.Client.Options;
+using MQTTnet.Client.Receiving;
+using MQTTnet.Extensions.ManagedClient;
+
+namespace Api.MQTT
+{
+    public class MqttService : BackgroundService
+    {
+        #region Events
+        public event EventHandler<MqttReceivedArgs>? MqttIsarMissionReceived;
+        public event EventHandler<MqttReceivedArgs>? MqttIsarTaskReceived;
+        public event EventHandler<MqttReceivedArgs>? MqttIsarStepReceived;
+        #endregion
+
+        public static MqttService? Instance { get; private set; }
+
+        private readonly ILogger<MqttService> _logger;
+
+        private readonly IManagedMqttClient _mqttClient;
+
+        private readonly ManagedMqttClientOptions _options;
+
+        private readonly TimeSpan _reconnectDelay = TimeSpan.FromSeconds(5);
+
+        public MqttService(ILogger<MqttService> logger, IConfiguration config)
+        {
+            Instance = this;
+            _logger = logger;
+            var mqttFactory = new MqttFactory();
+            _mqttClient = mqttFactory.CreateManagedMqttClient();
+
+            var mqttConfig = config.GetSection("Mqtt");
+
+            string username = mqttConfig.GetValue<string>("Username");
+            string password = config.GetValue<string>("mqtt-broker-password");
```

Comment:
The keyvault is defined in the config (appsettings.Development.json), currently set to flotilla keyvault.
1. Yes, good point
2. Yes, I can change it so that the app will only use keyvault if ASP.NET environment is set to production. We can always manually set this locally if we want to test the keyvault during development.

Reply:
Adressed issues, please see updated code